docs/testing-overview.md

```diff
@@ -0,0 +1,163 @@
+
+## Snapshot testing
+
+This is an overview of [snapshot testing] and how to best leverage snapshot tests.
+
+### TL;DR Broken snapshots
+
+When a snapshot test fails, it just means that a component's rendering has changed. If that was
+unintended, then the snapshot test just prevented a bug 😊
+
+However, if the change was intentional, follow these steps to update the snapshot:
+
+1. Run the following to update the snapshots:
+   ```sh
+   # --testPathPattern is optional but will be much faster by only running matching tests
+   npm run test-unit -- --updateSnapshot --testPathPattern path/to/tests
+   ```
+1. Review the diff and ensure the changes are expected and intentional
+1. Commit
+
+### What are snapshots?
+
+Snapshots are just a representation of some data structure generated by tests. Snapshots are stored
+in files and committed alongside the tests. When the tests are run, the data structure generated is
+compared with the snapshot on file.
+
+It's very easy to make a snapshot:
+
+```js
+test( 'foobar test', () => {
+  const foobar = { foo: 'bar' };
+
+  expect( foobar ).toMatchSnapshot();
+} );
+```
+
+This is the produced snapshot:
+
+```js
+exports[`test foobar test 1`] = `
+  Object {
+    "foo": "bar",
+  }
+`;
+```
+
+You should never create or modify a snapshot directly, they are generated and updated by tests.
+
+### Advantages
+
+* Trivial and concise to add tests
+* Protect against unintentional changes
+* Simple to work with
+* Reveal internal structures without running the application
+
+### Disadvantages
+
+* Not expressive
+* Only catch issues when changes are introduced
+* Are problematic for anything non-deterministic
+
+### Use cases
+
+Snapshot are mostly targeted at component testing. They make us conscious of changes to a
+component's structure which makes them _ideal_ for refactoring. If a snapshot is kept up to date
+over the course of a series of commits, the snapshot diffs record the evolution of a component's
+structure. Pretty cool 😎
+
+```js
+import { shallow } from 'enzyme';
+import SolarSystem from 'solar-system';
+import { Mars } from 'planets';
+
+describe( 'SolarSystem', () => {
+  test( 'should render', () => {
+    const wrapper = shallow( <SolarSystem /> );
+
+    expect( wrapper ).toMatchSnapshot();
+  } );
+
+  test( 'should contain mars if planets is true', () => {
+    const wrapper = shallow( <SolarSystem planets /> );
+
+    expect( wrapper ).toMatchSnapshot();
+    expect( wrapper.find( Mars ) ).toHaveLength( 1 );
+  } );
+} );
+```
+
+Reducer tests are also be a great fit for snapshots. They are often large, complex data structures
+that shouldn't change unexpectedly, exactly what snapshots excel at!
+
+### Working with snapshots
+
+You might be blindsided by CI tests failing when snapshots don't match. You'll need to
+[update snapshots] if the changes are expected. The quick and dirty solution is to invoke Jest with
+`--updateSnapshot`. In Calypso you can do that as follows:
```

Comment:
We're not "In Calypso" 😬

Reply:
😬 Thanks for spotting that… PR incoming.

Reply:
😎 we have one more PR open with docs ported from Calypso. I’ll make sure we don’t repeat the same mistake 😃